docs/docs/handling-events.md

```diff
@@ -2,7 +2,115 @@
 id: handling-events
 title: Handling Events
 permalink: docs/handling-events.html
-prev: state-and-lifecycle.html
 ---
 
-TODO
+Handling events with React elements is very similar to handling events on DOM elements. There are some syntactic differences:
+
+* React events are named using camelCase, rather than lowercase
+* With JSX you pass a function as the event handler, rather than a string
+
+For example, the HTML:
+
+```html
+<button onclick="activateLasers()">Activate Lasers</button>
+```
+
+is slightly different in React:
+
+```js
+<button onClick={activateLasers}>Activate Lasers</button>
+```
+
+Another difference is that you cannot return `false` to prevent default behavior in React. You must call `preventDefault` explicitly. For example, with plain HTML to prevent the default link behavior this works:
+
+```html
+<a href="#" onclick="console.log('the link was clicked'); return false">
+  click me
+</a>
+```
+
+In React, this would instead be:
+
+```js
+function handleClick(e) {
+  console.log('the link was clicked');
+  e.preventDefault();
+}
+
+function CustomLink() {
+  return <a href="#" onClick={handleClick}>click me</a>;
+}
+```
+
+React knows how to bubble and capture events according to the [W3C spec](https://www.w3.org/TR/DOM-Level-3-Events/), so you don't need to worry about cross-browser compatibility.
+
+When using React you should generally not need to call `addEventListener` to add listeners to a DOM element after it is created. Instead, just provide a listener when the element is initially rendered.
+
+A common pattern is for an event handler to be a method on the React component. For example, this `Toggle` component renders a button that lets the user toggle between "ON" and "OFF" states:
```

Comment:
>A common pattern is for an event handler to be a method on the React component.

Maybe "When you define component as a class, it is common to ..."

It might be confusing how a functional component could have methods.